### ISSUE TITLE
Database.all not running callback function?

### ISSUE THREAD

**Issue by kwfredeen** [mapbox/node-sqlite3#1211]
I am attempting to get a list of students who need a ride from a database. however, after I run the following code
```
const db = new sqlite3.Database('./database.db');

var studentList;
db.all('SELECT name, number, location FROM students WHERE claimed=0;', [], function(err, rows) {
    console.log('in callback');
    console.log('rows:'+rows);
    if(err==null){
        studentList=rows;
    } else {
        console.error(err);
    }
});
```
`studentList` is undefined, and neither "in callback" or the rows log appear in the console. `students` has rows that should be returned by the query.

Other queries run fine, so I'm assuming something I'm doing is causing the callback to be skipped.  I've looked for solutions in many places but have failed to find one.

**Comment by kwfredeen** [comment]
I figured out that it had to do with sqlite3 doing things asynchronously. I end up using studentList (in code I probably should've included honestly) before the callback has a chance to set it. 

I solved my problem by switching to sqlite-sync, but is there a way to do it without changing modules? Should I have moved some of the processing code into the callback function? I feel that the same race condition would occur if I did that.

**Comment by Lip-ee** [comment]
did you find a way to fix that, bro? I'm here with the same issue..

**Comment by stefnotch** [comment]
@Lip-ee If you've got the same issue, would it be possible for you to put a reasonably minimal project that reproduces this issue online?

**Comment by Lip-ee** [comment]
yeah, I've got the same problem, but now I forgot how I solved this lol. If I'm not wrong, was something about the path to my database.db file.
